src/app/screens/ConfirmPayment/index.tsx

```diff
@@ -42,12 +42,26 @@ function ConfirmPayment(props: Props) {
     ((invoice.satoshis || 0) * 10).toString()
   );
   const [fiatAmount, setFiatAmount] = useState("");
+  const [fiatBudgetAmount, setFiatBudgetAmount] = useState("");
 
   useEffect(() => {
     if (showFiat) {
       (async () => {
-        const res = await getFiatValue(budget);
-        setFiatAmount(res);
+        if (invoice.satoshis) {
+          const res = await getFiatValue(invoice.satoshis);
+          setFiatAmount(res);
+        }
+      })();
+    }
+  }, [invoice.satoshis, showFiat]);
```

Comment:
Combine the dependency check with

```suggestion
     if (showFiat && invoice.satoshis) {
      (async () => {
          const res = await getFiatValue(invoice.satoshis);
          setFiatAmount(res);
      })();
    }
  }, [invoice.satoshis, showFiat]);
```

Reply:
lol, true